### PR TITLE
Get canvas example working

### PR DESCRIPTION
**Pull request by fitzgen**
Fixes https://github.com/rustwasm/wasm-bindgen/issues/444

### CODE REVIEW

File: crates/web-sys/webidls/enabled/CanvasRenderingContext2D.webidl [hunk 1]
```diff
@@ -151,20 +149,18 @@ interface CanvasTransform {
...
 
 [NoInterfaceObject]
-interface CanvasCompositing {
+interface mixin CanvasCompositing {
   attribute unrestricted double globalAlpha; // (default 1.0)
   [Throws]
   attribute DOMString globalCompositeOperation; // (default source-over)
 };
 
-[NoInterfaceObject]
```
Comment (alexcrichton) [inline crates/web-sys/webidls/enabled/CanvasRenderingContext2D.webidl]: Is this a transformation/interpretation that we should bake directly into the webidl parsing? Basically interpreting any interface tagged as `[NoInterfaceObject]` as a mixin instead?

File: crates/web-sys/webidls/enabled/CanvasRenderingContext2D.webidl [hunk 2]
```diff
@@ -107,33 +107,31 @@ interface CanvasRenderingContext2D {
...
-CanvasRenderingContext2D implements CanvasRect;
-CanvasRenderingContext2D implements CanvasDrawPath;
-CanvasRenderingContext2D implements CanvasUserInterface;
-CanvasRenderingContext2D implements CanvasText;
-CanvasRenderingContext2D implements CanvasDrawImage;
-CanvasRenderingContext2D implements CanvasImageData;
-CanvasRenderingContext2D implements CanvasPathDrawingStyles;
-CanvasRenderingContext2D implements CanvasTextDrawingStyles;
-CanvasRenderingContext2D implements CanvasPathMethods;
-CanvasRenderingContext2D implements CanvasHitRegions;
```
Comment (alexcrichton) [inline crates/web-sys/webidls/enabled/CanvasRenderingContext2D.webidl]: Could we interpret `A implements B` as `A includes B` in webidl parsing? I actually have no idea what the difference is between the two...

File: examples/canvas/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "wasm-bindgen-canvas-demo"
+version = "0.1.0"
+authors = ["Nick Fitzgerald <fitzgen@gmail.com>"]
```
Comment (alexcrichton) [inline examples/canvas/Cargo.toml]: s/you/"The wasm-bindgen Developers"/

File: examples/canvas/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Closure examples
```
Comment (alexcrichton) [inline examples/canvas/README.md]: copy/paste

File: crates/web-sys/webidls/enabled/CanvasRenderingContext2D.webidl (hunk 2 above)
Comment (fitzgen) [reply crates/web-sys/webidls/enabled/CanvasRenderingContext2D.webidl]: We could but (a) it isn't standard webidl, and (b) there isn't any of that anymore after this PR :-p

Comment (alexcrichton) [reply crates/web-sys/webidls/enabled/CanvasRenderingContext2D.webidl]: Heh ok!

**Pull request merged**